src/v2/ion/ViewClassNamesFactory.js

```diff
@@ -0,0 +1,88 @@
+/* eslint complexity: [2, 25] */
+import {FORMS} from './RemediationConstants';
+
+const FORMNAME_CLASSNAME_MAPPINGS = {
+  [FORMS.IDENTIFY]: {
+    'identify': ['identify', 'primary-auth'],
+    'password': ['identify', 'primary-auth']
+  },
+  [FORMS.ENROLL_PROFILE]: {
+    'enroll-profile': ['enroll-profile', 'registration'],
+  },
+  [FORMS.CHALLENGE_AUTHENTICATOR]: {
+    email: ['challenge-authenticator-email', 'mfa-verify-passcode'],
+    password: ['challenge-authenticator-password', 'mfa-verify-password'],
+    sms: ['challenge-authenticator-voice', 'mfa-verify-passcode'],
+    voice: ['challenge-authenticator-voice', 'mfa-verify-passcode'],
+    'security_question': ['challenge-authenticator-security_question', 'mfa-verify-question']
+  },
+  [FORMS.ENROLL_AUTHENTICATOR]: {
+    email: ['enroll-authenticator-email', 'enroll-email'],
+    password: ['enroll-authenticator-password', 'enroll-password'],
+    sms: ['enroll-authenticator-sms', 'enroll-sms'],
+    voice: ['challenge-authenticator-voice', 'mfa-verify-passcode'],
+    'security_question': ['enroll-authenticator-security_question', 'enroll-question'],
+  },
+  [FORMS.AUTHENTICATOR_ENROLLMENT_DATA]: {
+    voice: ['authenticator-enrollment-data'],
+    sms: ['authenticator-enrollment-data']
+  },
+  [FORMS.AUTHENTICATOR_VERIFICATION_DATA]: {
+    voice: ['authenticator-verification-data'],
+    sms: ['authenticator-verification-data']
+  },
+  [FORMS.SELECT_AUTHENTICATOR_ENROLL]: {
+    'select-authenticator-enroll': ['select-authenticator-enroll', 'enroll-choices']
+  },
+  [FORMS.SELECT_AUTHENTICATOR_AUTHENTICATE]: {
+    'password': ['select-authenticator-authenticate']
+  },
+  [FORMS.REENROLL_AUTHENTICATOR]: {
+    'password':['reenroll-authenticator', 'password-expired']
+  },
+  [FORMS.REENROLL_AUTHENTICATOR_WARNING]: {
+    'password':['reenroll-authenticator_warning']
+  },
+  [FORMS.RESET_AUTHENTICATOR]: {
+    'password':['reset-authenticator', 'forgot-password']
+  },
+};
+
+const getClassNameMapping = (appState) => {
+  const formName = appState.get('currentFormName');
+  const authenticator = appState.get('currentAuthenticatorEnrollment') || appState.get('currentAuthenticator');
+  let authenticatorType = authenticator && authenticator.type;
+  const methodType = authenticator && authenticator.methods && authenticator.methods[0].type;
+  const isPasswordRecoveryFlow = appState.get('recoveryAuthenticator') &&
+    appState.get('recoveryAuthenticator').type === 'password';
+  let type;
+
+  if (authenticatorType === 'phone') {
+    /**
+      Both sms and call have same type phone
+      currentAuthenticatorEnrollment is during verify and currentAuthenticator during enroll flows
+    **/
+    type = `${methodType}`;
+  }
+  else if (authenticatorType) {
+    type = `${authenticatorType}`;
+  }
+
```

Comment:
Wonder if 70-82 can be

```suggestion
cssNames = either FORMNAME_CLASSNAME_MAPPINGS[formName][type] or FORMNAME_CLASSNAME_MAPPINGS[formName][formName];

if is password recovery then
  cssNames.push('forgot-password')
endif

```

Reply:
can be simplified to if use `_` as default type.
```
let type = '_';
if authenticator exists then
  type = authenticatorType;

if authenticator == phone then
  type = methodType;

cssNames = FORMNAME_CLASSNAME_MAPPINGS[formName][type];

if is password recovery then
  cssNames.push('forgot-password')
endif
```

Reply:
Right, we would still eventually return` FORMNAME_CLASSNAME_MAPPINGS[formName][type]` ?

For forms like `success`, `terminal` and a few others without type, the type would evaluate to `_` which would then mean that we need to add an entry 

```
  [FORMS.SUCCESS]: {
    '_': ['success']
  },
```
I am trying to avoid adding an entry for each form and doing `FORMNAME_CLASSNAME_MAPPINGS[formName] || [formName];` which would return ['success'] if it doesn't find an entry for a form.

Reply:
yes, if we omit V2 form that has no corresponding to V1,  it could be
```
v2ClassName = formName;
if type != '_' 
  v2ClassName = formName + '--' + type;

v1Classes = [];
if FORMNAME_CLASSNAME_MAPPINGS[formName]
  v1Classes = FORMNAME_CLASSNAME_MAPPINGS[formName][type];

cssNames = v1Classes.concat(v2ClassName);
```

Reply:
IMO this feels like someone has to look at this code and figure out what classes are being added and my goal was to to make it so that someone could just look at the mapping and see that the classnames are being applied with an exception that if an entry does not exist we will add the formName as the class name as default.